src/devtoolsPanel.ts

```diff
@@ -311,35 +312,42 @@ export class DevToolsPanel {
         // Convert the local url to a workspace path
         const transformer = new debugCore.UrlPathTransformer();
         void transformer.launch({ pathMapping: this.config.pathMapping });
-        const localSource = { path: sourcePath };
-        await transformer.fixSource(localSource);
 
-        sourcePath = localSource.path || sourcePath;
+        // origin in this case is trivial since we expect fixSource to take it out
+        const localSource = { path: sourcePath, origin: 'vscode-webview://' };
+        await transformer.fixSource(localSource);
 
-        // Convert the workspace path into a VS Code url
-        let uri: vscode.Uri | undefined;
-        try {
-            if (vscode.env.remoteName) {
-                uri = vscode.Uri.parse(sourcePath, true);
-            } else {
-                uri = vscode.Uri.file(sourcePath);
-            }
-        } catch {
-            uri = undefined;
+        // per documentation if the file was correctly resolved origin will be cleared.
+        // https://github.com/Microsoft/vscode-chrome-debug-core/blob/main/src/transformers/urlPathTransformer.ts
+        if (!localSource.origin) {
+            // Convert the workspace path into a VS Code url
+            const uri = vscode.Uri.file(localSource.path);
+            await this.openInEditor(uri, line, column);
+        } else {
+            await ErrorReporter.showInformationDialog({
+                errorCode: ErrorCodes.Error,
+                title: 'Error while opening file in editor.',
```

Comment:
I actually wouldn't put "Error" in the title here, makes me think something is wrong when in fact it is working like expected. i'd maybe say something like
'Unable to open file in editor'
'${sourcePath} does not map to a local file.'

Reply:
thanks for all the feedback @bgoddar